File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        counter = prefix = 0
        n = len(nums)
        store = defaultdict(int)
        store[0] = 1
        for num in nums:
            prefix += num
            if prefix-k in store:
                counter += store[prefix-k]
            store[prefix] += 1
            
        return counter

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int n = nums.size(), cur = 0, cnt = 0;
        unordered_map<int, int> m{{0,1}};
        
        for (auto x: nums) {
            cur += x;
            cnt += m[cur - k];
            m[cur]++;
        }
        return cnt;
    }
};